src/plugins/data/public/antlr/shared/utils.ts

```diff
@@ -93,24 +95,27 @@ export const fetchData = (
   });
 };
 
-// Specific fetch function for table schemas
-// TODO: remove this after using data set table schema fetcher
-export const fetchTableSchemas = (tables: string[], api: any, queryString: QueryStringContract) => {
-  return fetchData(
-    tables,
-    (table, dataSourceId, title) => ({
-      query: { query: `DESCRIBE TABLES LIKE ${table}`, format: 'jdbc' },
-      df: {
-        meta: {
-          queryConfig: {
-            dataSourceId: dataSourceId || undefined,
-            title: title || undefined,
-          },
-        },
+export const fetchColumnValues = (
+  tables: string[],
+  column: string,
+  services: IDataPluginServices,
+  dataset?: Dataset
```

Comment:
Why is `dataset` optional here? Would it still work without it?

Reply:
The original reason for this is the dataset would be used solely for mds, if there is another cluster then it will pass in the dataset to use the same data source that is already engaged. Also, the dataset being passed into the getSuggestion functions don't require dataset, so its type is `Dataset | undefined` before calling this fetchColumnValues helper function. 
Should I change this to have dataset be required?